lib/PlatoonService/src/PlatoonController.cpp

```diff
@@ -157,8 +155,26 @@ void PlatoonController::process()
         }
     }
 
+    /* Send current position as waypoint in a constant distance interval. */
+    if (WAYPOINT_DISTANCE_INTERVAL < PlatoonUtils::calculateAbsoluteDistance(m_lastSentWaypoint, m_currentVehicleData))
+    {
+        if (false == m_outputWaypointCallback(m_currentVehicleData))
+        {
+            LOG_ERROR("Failed to send waypoint to next vehicle.");
+        }
+        else
+        {
+            m_lastSentWaypoint = m_currentVehicleData;
+        }
+    }
+
+    /* Are there enough waypoints available at the start of the drive? */
+    if ((false == m_processingChainRelease) && (MIN_AVAILABLE_WAYPOINTS < numberOfAvailableWaypoints))
+    {
+        m_processingChainRelease = true;
+    }
     /* Process chain on timeout. */
-    if ((true == m_processingChainTimer.isTimeout()) && (nullptr != m_motorSetpointCallback))
+    else if ((true == m_processingChainTimer.isTimeout()) && (nullptr != m_motorSetpointCallback))
```

Comment:
Why checking for m_motorSetpointCallback? The comment talks only about the timeout.